common/src/com/thoughtworks/go/security/X509CertificateGenerator.java

```diff
@@ -64,8 +66,13 @@ public X509CertificateGenerator() {
 
     public void createAndStoreX509Certificates(File keystore, File truststore, File agentKeystore,
                                       String password, String principalDn) {
-        storeX509Certificate(keystore, password, createCertificateWithDn(principalDn));
-        storeX509Certificate(truststore, password, createAndStoreCACertificates(agentKeystore));
+        if (!keystore.exists()) {
```

Comment:
I tried deleting the keystore and restarted the server. The agent continued to ping without any issues.